service/worker/scheduler/workflow.go

```diff
@@ -246,18 +248,29 @@ func (s *scheduler) run() error {
 		for s.processBuffer() {
 		}
 		s.updateMemoAndSearchAttributes()
+		// if run out of actions or no more jobs to run, finish the schedule workflow
+		for s.hasMinVersion(FinishIdleSchedule) && (nextWakeup.IsZero() || (!s.Schedule.State.Paused && !s.canTakeScheduledAction(false, false))) {
+			// handle signals before closing
+			scheduleChanged = s.processSignals()
```

Comment:
I don't think it's possible for there to be more signals here, is it? or at least, it's not possible for processSignals to do anything here, because we just called it above and we haven't waited on any signal channels in between. (processSignals doesn't drain signal channels, it only processes the ones we've received already)

if we do get a signal before closing, the wft will be rejected and it'll do a new one, essentially restarting from the previous sleep. so I think it shouldn't need an explicit drain, but it would be good to test if possible. in any case, c-a-n has the same issue so if we need an explicit drain it should be there too.

Reply:
I see, if wft is rejected if we get a signal then this should not be necessary. let me see if I can test this.